http.go

```diff
@@ -100,7 +67,73 @@ func validateAPIKey(cert *x509.Certificate, apiKeyStr string) bool {
 	return true
 }
 
-func NewReverseProxy(config *Config, db *sql.DB) *httputil.ReverseProxy {
+type ExtraFees struct {
+	ID         string  `json:"id"`
+	Percentage float64 `json:"percentage"`
+}
+
+func addRequestExtraFees(body []byte, extraFee Fee) ([]byte, error) {
+	var jsonBody map[string]interface{}
+	if err := json.Unmarshal(body, &jsonBody); err != nil {
+		return nil, fmt.Errorf("error unmarshaling JSON: %w", err)
+	}
+
+	jsonBody["extraFees"] = ExtraFees{
+		ID:         strconv.Itoa(extraFee.PartnerID),
+		Percentage: extraFee.FeePercentage,
+	}
+
+	modifiedBody, err := json.Marshal(jsonBody)
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling JSON: %w", err)
+	}
+
+	return modifiedBody, nil
+}
+
+func modifyResponsePercentages(data interface{}, extraFeePercentage float64) interface{} {
+	// The response is a map of currency pairs
+	currencyMap, ok := data.(map[string]interface{})
+	if !ok {
+		return data
+	}
+
+	// Iterate through each currency pair
+	for _, pairData := range currencyMap {
+		pairMap, ok := pairData.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		// Each pair has another map of currency data
+		for _, currencyData := range pairMap {
+			currencyMap, ok := currencyData.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			// Check for fees.percentage
+			if fees, ok := currencyMap["fees"].(map[string]interface{}); ok {
+				if percentage, ok := fees["percentage"].(float64); ok {
+					fees["percentage"] = percentage + extraFeePercentage
```

Comment:
Would it help to convert the floats to ints when calculating the percentage (and back to float) to avoid floating point issues?